rust/driver-ipc/src/driver_client.rs

```diff
@@ -1,78 +1,77 @@
-use std::{collections::HashSet, sync::OnceLock};
+use std::collections::HashSet;
 
-use tokio::{
-    sync::mpsc::{channel, Sender},
-    task,
-};
+use tokio::{sync::watch, task};
+use tokio_stream::{Stream, StreamExt};
 
-use crate::{Client, ClientError, EventCommand, Id, IpcError, Mode, Monitor, ReplyCommand, Result};
-
-// used to terminate existing receivers
-static RECEIVER_SHUTDOWN_TOKEN: std::sync::Mutex<OnceLock<Sender<()>>> =
-    std::sync::Mutex::new(OnceLock::new());
+use crate::*;
 
 /// Extra API over Client which allows nice fancy things
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
I was thinking, it could be nice to be able to clone the client and work on different states simultaniously, but it might be missleading to use `clone` for this purpose? If we change this, there will be no possibility to do so. 

A usecase might be to "backup" the current state, make changes to the driver and afterwarts revert to the backup.

Reply:
Yeah. I'd rather support only one case (independent states or synced states), since two would just add to confusion. Your use-case is better in the end as well, so my vote would be to not support synced states 

It's still not too clear whether clone would sync states across clients or keep them separate however, so maybe this would be better as a separate method with doc comment to make it clear

Reply:
Resolved in 1985a4094028d01c054fc9fe7725c4e939cacdc3

Reply:
> Resolved in [1985a40](https://github.com/MolotovCherry/virtual-display-rs/commit/1985a4094028d01c054fc9fe7725c4e939cacdc3)

Lgtm!